.github/workflows/build-and-deploy-container-image.yaml

```
@@ -0,0 +1,403 @@
+        description: An SSH Key to load the repo https://github.com/neohelden/global-audit-ci-config
+        required: true
+      argocd-auth-token:
+        description: Token for authenticating on ArgoCD host
+        required: true
+      gh-app-id:
+        description: Id of a Github App being used for triggering other workflows
+        required: true
+      gh-app-private-key:
+        description: Private key of a Github App being used for triggering other workflows
+        required: true
+      github-token:
+        description: Token used for accessing GitHub.com
+        required: true
+
+jobs:
+
+  build-and-test:
+    name: build-and-test
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+
+    steps:
+      - uses: neohelden/actions-library/build-and-test-with-yarn@feat/reusable-workflows
+        with:
+          sonar-token: ${{ secrets.sonar-token }}
+          github-token: ${{ secrets.github-token }}
+          checkout-ref: ${{ inputs.checkout-ref }}
+          sonar-project-base-dir: ${{ inputs.sonar-project-base-dir }}
+          sonar-organization: ${{ inputs.sonar-organization }}
+          sonar-projectKey: ${{ inputs.sonar-projectKey }}
+          sonar-branch-name: ${{ inputs.sonar-branch-name }}
+          sonar-sources: ${{ inputs.sonar-sources }}
+          sonar-inclusions: ${{ inputs.sonar-inclusions }}
+          sonar-exclusions: ${{ inputs.sonar-exclusions }}
+          sonar-tests: ${{ inputs.sonar-tests }}
+          sonar-test-inclusions: ${{ inputs.sonar-test-inclusions }}
+          sonar-test-exclusions: ${{ inputs.sonar-test-exclusions }}
+          sonar-coverage-exclusions: ${{ inputs.sonar-coverage-exclusions }}
+          sonar-lcov-reportpaths: ${{ inputs.sonar-lcov-reportpaths }}
+          sonar-verbose: ${{ inputs.sonar-verbose }}
+          default-branch: ${{ inputs.default-branch }}
+          tsconfig-build-file: ${{ inputs.tsconfig-build-file }}
+          prettier-source-paths: ${{ inputs.prettier-source-paths }}
+          eslint-ext-patterns: ${{ inputs.eslint-ext-patterns }}
+          eslint-configuration-file: ${{ inputs.eslint-configuration-file }}
+          
+  check-code-compliance:
+    name: check-code-compliance
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+
+    steps:
+      - uses: neohelden/actions-library/check-code-compliance@feat/reusable-workflows
+        with: 
+          repo-ssh-key: ${{ secrets.global-audit-ci-repo-key }}
+          checkout-ref: ${{ inputs.checkout-ref }}
+          artifact-name: ${{ inputs.artifact-name }}
+          license-exclude-packages: ${{ inputs.license-exclude-packages }}
+          audit-ci-config: ${{ inputs.audit-ci-config }}
+          
+  build-docker-image:
+    name: build-docker-image
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    needs:
+      - build-and-test
+      - check-code-compliance
+
+    steps:
+      - uses: neohelden/actions-library/build-docker-image@feat/reusable-workflows
```

Wen need to change this to main
```suggestion
      - uses: neohelden/actions-library/build-docker-image@main
```